scripts/001-binutils-PPU.sh

```diff
@@ -1,19 +1,21 @@
 #!/bin/sh -e
+#
 # binutils-PPU.sh by Naomi Peori (naomi@peori.ca)
+# Modified by luizfernandonb (luizfernando.nb@outlook.com)
 
 BINUTILS="binutils-2.22"
 
 if [ ! -d ${BINUTILS} ]; then
 
   ## Download the source code.
-  if [ ! -f ${BINUTILS}.tar.bz2 ]; then wget --continue https://ftp.gnu.org/gnu/binutils/${BINUTILS}.tar.bz2; fi
+  if [ ! -f ${BINUTILS}.tar.gz ]; then wget --continue https://ftp.gnu.org/gnu/binutils/${BINUTILS}.tar.gz; fi
 
   ## Download an up-to-date config.guess and config.sub
   if [ ! -f config.guess ]; then wget --continue https://git.savannah.gnu.org/cgit/config.git/plain/config.guess; fi
   if [ ! -f config.sub ]; then wget --continue https://git.savannah.gnu.org/cgit/config.git/plain/config.sub; fi
 
   ## Unpack the source code.
-  tar xfvj ${BINUTILS}.tar.bz2
+  pigz -dc ${BINUTILS}.tar.gz | tar -xvf -
```

Comment:
And what in the world is `pigz`?  Only use tools people are likely to actually have...

Reply:
:+1:

Reply:
If you read the [README](https://github.com/ps3dev/ps3toolchain/pull/106/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R31) changes you know that `pigz` is in the libraries that are asked to install before compiling, I would not use a program in the script file without even installing it

Reply:
Well, I don't think making the build 8 seconds slower warrants adding the requirement to find and install a new tool.  (Not everyone runs Ubuntu or macOS.)

Reply:
You are looking at a simple file, you have to look at the entire toolchain, seeing this, it is clear that the extraction of `GCC 7.2.0` and `binutils 2.22` are the most time consuming of the entire toolchain, and certainly a program to extract these files using multiple threads will make a good difference in the complete installation time of the toolchain

My PC has a Pentium E6700 from 2010, so it doesn't make much difference, but for those who extract on an i7 or i5, you may notice the difference

Reply:
Huh?  _Compiling_ gcc is what takes time.  Unpacking has got to be <1% of the entire toolchain build time.  If you have to dig up a nonstandard tool and compile and install it, that's going to take several minutes, which is unlikely to make it a win even if your computer is really slow and your internet connection is really fast.

On my computer (POWER9) the extraction of `gcc-7.2.0.tar.xz` takes 6 seconds.  CPU load is 131.3%, so one hardware thread is running `xz`, one is running `tar`, and the remaining 62 are idle.  Now the _compliation_ of gcc, that takes 4 minutes 42 seconds even though it can use all the threads.  So of the total time for extracting and compiling gcc, the extracting amounted for 2% of the time.  This ratio would stay the same even with a slower CPU (because both unpacking and compilation would be slower), but on a CPU with fewer threads the extraction would amount to even less because the compilation would be slower but not the extraction (which can only use 2 threads anyway).  So assuming the CPU has only 32 or less threads I'd say my guesstimate of <1% hits the mark.  😄 

Downloading gcc as `.gz` takes me 3 minutes 24 seconds.  Downloading gcc as `.xz` takes 1 minute 55 seconds.  So it saves more than 40% of the download time, or a minute and a half in absolute terms.  Yes, this is over an IPv6 tunnel, but if we assume slow computers we should assume slow connections too (and not necessarily in combination).

At the end of the day I guess your mileage will vary, but that's also why I wouldn't accept this as a mandatory one-size-fits all solution.  If you were to make it as an option for those who would both actually benefit and care enough to invest in getting this tool, then maybe...

Reply:
Ok, I will remove the `pigz` from the source code, I ended up agreeing with your explanation

Reply:
But we also have to look at people who do not have a PC as good as yours, here, unpacking the GCC takes a long time, and the compilation too, adding the unpacking/compilation for the PPU and SPU, here it takes at least 2 hours to complete (without taking into account binutils and GDB)

Reply:
Done, if you can review this PR, I will be grateful 🤝

Reply:
Out of interrest, did you also measure how much of those 2 hours is spent on the actual uncompression?  With and without the pigz?  If you measure using `time` be careful that it only measures one process and that you have a pipeline in the pigz case but not in the regular one.  To get a fair measurement you can change the regular case to also be a pipeline and put a `time` on the `xz -dc` process to race it against `pigz -dc`.